CodeEdit/Features/NavigatorArea/SourceControlNavigator/SourceControlNavigatorView.swift

```diff
@@ -10,39 +10,65 @@ import SwiftUI
 struct SourceControlNavigatorView: View {
     @EnvironmentObject private var workspace: WorkspaceDocument
 
+    var body: some View {
+        VStack(spacing: 0) {
+            if let sourceControlManager = workspace.workspaceFileManager?.sourceControlManager {
+                SourcControlNavigatorTabs()
+                    .environmentObject(sourceControlManager)
+                    .onAppear {
+                        sourceControlManager.startPeriodicFetch(interval: 10)
+                    }
+                    .onDisappear {
+                        sourceControlManager.stopPeriodicFetch()
+                    }
+            }
+        }
+        .safeAreaInset(edge: .bottom, spacing: 0) {
+            SourceControlNavigatorToolbarBottom()
+        }
+    }
+}
+
+struct SourcControlNavigatorTabs: View {
+    @EnvironmentObject var sourceControlManager: SourceControlManager
     @State private var selectedSection: Int = 0
 
     var body: some View {
-        VStack {
+        if sourceControlManager.isGitRepository {
             SegmentedControl(
                 $selectedSection,
                 options: ["Changes", "Repositories"],
                 prominent: true
             )
             .frame(maxWidth: .infinity)
-            .frame(height: 27)
+            .frame(height: 26)
             .padding(.horizontal, 8)
-            .padding(.bottom, 2)
-            .overlay(alignment: .bottom) {
-                Divider()
-            }
-
-            if let sourceControlManager = workspace.workspaceFileManager?.sourceControlManager {
-                if selectedSection == 0 {
-                    SourceControlNavigatorChangesView(
-                        sourceControlManager: sourceControlManager
-                    )
-                }
-
-                if selectedSection == 1 {
-                    SourceControlNavigatorRepositoriesView(
-                        sourceControlManager: sourceControlManager
-                    )
+            .task {
+                Task {
+                    try await sourceControlManager.refreshRemotes()
+                    try await sourceControlManager.refreshStashEntries()
                 }
             }
```

Comment:
Task within a task modifier is redundant. Without the inner task we'll just need to be explicit about discarding or handling errors from here.
```suggestion
            .task {
                do {
                    try await sourceControlManager.refreshRemotes()
                    try await sourceControlManager.refreshStashEntries()
                } catch {
                    // TODO: Handle errors
                }
            }
```